tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+
+## 10,000 foot view
+Girder Worker is an application that wraps Celery; it relies on Celery's distributed task execution mechanisms to do work, while reflecting task state and logging information back into [Girder](https://girder.readthedocs.io/) for visualization, debugging and management. Girder Worker is designed to receive and execute tasks that have been produced by Girder. Messages that encode what tasks and task arguments are passed from Girder to Girder Worker via a separate service called a [broker](http://docs.celeryproject.org/en/latest/getting-started/brokers/). In many common setups this broker is [RabbitMQ](https://www.rabbitmq.com/). 
+
+Girder Worker runs in a separate process from Girder. It may run on the same physical machine as Girder,  or may be run on one or more separate machines. At a low level, RabbitMQ brokers the communication between Girder and Girder Worker.  If Girder and Girder Worker are running on separate machines, both machines must be able to connect to the server running RabbitMQ.
+
+Assuming you have configured girder-worker to connect to the RabbitMQ service, running Girder Worker is easy. From a command line simply run:
+
+```python
+$> girder-worker
+```
+
+Or if you are using [systemd](https://www.freedesktop.org/wiki/Software/systemd/) you may create a simple service file:
+
+```sh
+[Unit]
+Description=Girder Worker Service
+After=network.target
+
+[Service]
+Type=simple
+User=GIRDER_USER
+Group=GIRDER_GROUP
+
+ExecStart=/path/to/girder-worker
+ExecStop=/path/to/bin/celery multi stopwait worker
+
+[Install]
+WantedBy=multi-user.target
+```
+Replace ```GIRDER_USER``` and ```GIRDER_GROUP``` with the user and group you want the service to run as. Save the file as ```/etc/systemd/system/girder_worker.service``` and run ```sudo systemctl daemon-reload```. Finally execute ```systemctl start girder_worker``` to run the girder_worker service.
+
+### Vocabulary
+
+Vocabulary can be an area of confusion when dealing with these systems. In an attempt to be clear we will draw a strong, jargon-like, distinction between the terms *Job* and *Task*.
+
+The term **Job** will always refer to the concept of a Girder [Job](http://girder.readthedocs.io/en/latest/plugins.html#jobs). Jobs are displayed on the Job's panel in Girder. They are represented in the database as [Job Models](https://github.com/girder/girder/blob/master/plugins/jobs/server/models/job.py). Girder Job's have various states and notification streams, and can be scheduled to execute on Girder Worker. 
+
+The term **Task** on the other hand will always refer to the Celery concept of a [Task](http://docs.celeryproject.org/en/latest/userguide/tasks.html). Tasks are typified by decorated python functions that can be scheduled to run on Girder Worker via the Celery [task calling api](http://docs.celeryproject.org/en/latest/userguide/calling.html). They are not persisted in any database but exist as python objects which are imported from a package. 
```

Comment:
I'm adding some extra text here,  but just to clarify - I'm talking specifically about the Celery concept of a Task. This means a function that is decorated by ```@app.task```. This turns the function into a Python object (referred to as the 'task object'). That means the task doesn't exist until the python module it is apart of is executed (e.g. on import).